fix: improve treesitter utilities to handle markdown code blocks

oddish3 opened the pull request:
Hi!
Thought I would make a PR incase anybody wanted to be able to type latex math and include inline R code at the same time, in both markdown and Quarto via nvim.
Changes:
- Add detection for code blocks in markdown files (when using treesitter afaik)
- Ensure math snippets don't trigger after code blocks with dollar signs (eg R lang)

before fix:
$x^2$ <- works fine here
```{r} 
mydata$variable
```
 mkxsr <- doesnt work here
after fix:
$x^2$ <- works fine here
```{r} 
mydata$variable
```
 $x^2$ <- works fine here
 
 In case you are wondering this is valid syntax in the R language!

iurimateus commented on lua/luasnip-latex-snippets/util/ts_utils.lua:
```diff
@@ -1,49 +1,79 @@
+    if current_filetype == "markdown" then
+        local block_node = node
+        while block_node do
+            if CODE_BLOCK_NODES[block_node:type()] then
```
Can we remove the `filetype == "markdown"` guard? i.e. any downsides/side effects on considering `CODE_BLOCK_NODES` as non-math nodes in `tex` files?

oddish3 replied:
The filetype guard can be safely removed. 
The CODE_BLOCK_NODES types (fenced_code_block and indented_code_block) are specific to markdown-style syntax and won't match against LaTeX code block environments, which use different node types. 
Removing the guard would just add a quick no-op traversal for LaTeX files with no functional impact.

oddish3 commented:
You're right - I've removed the filetype guard. 
The CODE_BLOCK_NODES types are specific to markdown-style syntax and won't match against LaTeX code block environments (which use different node types).
 The only impact is an additional quick traversal that will be a no-op for LaTeX files since no matching code block nodes will be found. 
I've updated the PR with the simplified version
